runhouse/resources/module.py

```diff
@@ -49,6 +49,24 @@
 ]
 
 
+def _env_for_module(passed_in_env=None, system=None):
+    if passed_in_env is not None:
+        return _get_env_from(passed_in_env)
+
+    active_env_servlet_env = obj_store.get_process_env()
+    if active_env_servlet_env:
+        return active_env_servlet_env
+
+    if system and isinstance(system, Cluster):
+        return system.default_env
+
+    potential_default_env = _get_env_from(_default_env_if_on_cluster())
+    if potential_default_env:
+        return potential_default_env
+
+    return Env()
```

Comment:
I'm not sure about this either - if we don't have an env to infer, I think we should just fall back to the cluster constructor's behavior, not set a default or placeholder.